aztec/src/main/kotlin/org/wordpress/aztec/watchers/DeleteMediaElementWatcher.kt

```diff
@@ -0,0 +1,55 @@
+package org.wordpress.aztec.watchers
+
+import android.text.Editable
+import android.text.TextWatcher
+import org.wordpress.aztec.AztecText
+import org.wordpress.aztec.Constants
+import org.wordpress.aztec.spans.AztecMediaSpan
+import java.lang.ref.WeakReference
+
+class DeleteMediaElementWatcher(aztecText: AztecText) : TextWatcher {
+
+    private val aztecTextRef: WeakReference<AztecText?> = WeakReference(aztecText)
+
+    override fun beforeTextChanged(text: CharSequence, start: Int, count: Int, after: Int) {
+        if (aztecTextRef.get()?.isTextChangedListenerDisabled() ?: true) {
+            return
+        }
+
+        var deletedMedia = count > 0  && containsMediaChars(text, start, count)
+
+        if (deletedMedia) {
+            var mediaSpanList = findAllMediaSpansWithinRange(start, count)
```

Comment:
sure - wanted to have that separated but it surely looks slimmer this way. Addressed in 02c1723